src/store/mmap.rs

```diff
@@ -239,13 +241,9 @@ impl<E: Element> Store<E> for MmapStore<E> {
     }
 
     fn compact(&mut self, _config: StoreConfig, _store_version: u32) -> Result<bool> {
-        if self.map.is_none() {
-            return Ok(false);
-        }
-
-        self.map = None;
+        let map = self.map.take();
```

Comment:
I hadn't run across this yet, but if I understand correctly, this drops the reference by locally assigning it so that on return it's dropped?

Reply:
`take` does the following

```
let mut my_option = Some("hello");
let rest = my_option.take();
assert_eq!(rest, Some("hello"));
assert_eq!(my_option, None);
```

So we pull it out, and then check if it was a thing or not when returning, and then it gets dropped.